aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AmazonPinpointAnalyticsPlugin.java

```diff
@@ -18,35 +18,58 @@
 import android.app.Application;
 import android.content.Context;
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.StringDef;
 
+import com.amplifyframework.analytics.AnalyticsChannelEventName;
 import com.amplifyframework.analytics.AnalyticsEvent;
 import com.amplifyframework.analytics.AnalyticsException;
 import com.amplifyframework.analytics.AnalyticsPlugin;
-import com.amplifyframework.analytics.AnalyticsProfile;
 import com.amplifyframework.analytics.Properties;
 import com.amplifyframework.analytics.Property;
+import com.amplifyframework.analytics.UserProfile;
 import com.amplifyframework.core.Amplify;
 import com.amplifyframework.hub.HubChannel;
 import com.amplifyframework.hub.HubEvent;
 
 import com.amazonaws.mobileconnectors.pinpoint.PinpointManager;
 import com.amazonaws.mobileconnectors.pinpoint.analytics.AnalyticsClient;
+import com.amazonaws.mobileconnectors.pinpoint.targeting.TargetingClient;
+import com.amazonaws.mobileconnectors.pinpoint.targeting.endpointProfile.EndpointProfile;
+import com.amazonaws.mobileconnectors.pinpoint.targeting.endpointProfile.EndpointProfileLocation;
+import com.amazonaws.mobileconnectors.pinpoint.targeting.endpointProfile.EndpointProfileUser;
 import com.amazonaws.regions.Regions;
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.Collections;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 
 /**
  * The plugin implementation for Amazon Pinpoint in Analytics category.
  */
 public final class AmazonPinpointAnalyticsPlugin extends AnalyticsPlugin<Object> {
+    @SuppressWarnings("checkstyle:WhitespaceAround")
+    @Retention(RetentionPolicy.SOURCE)
+    @StringDef({
+        USER_NAME,
+        USER_EMAIL,
+        USER_PLAN
+    })
+    private @interface PinpointUserProfileAttribute {}
+    private static final String USER_NAME = "name";
+    private static final String USER_EMAIL = "email";
+    private static final String USER_PLAN = "plan";
     private final Application application;
```

Comment:
nit: to help break up the groupings, add an empty line after the constants, before the `Application` declaration.